plugins/org.jboss.ide.eclipse.freemarker/src/org/jboss/ide/eclipse/freemarker/editor/Editor.java

```diff
@@ -492,48 +490,97 @@ public void run() {
 					}
 					getFile().deleteMarkers(IMarker.PROBLEM, true,
 							IResource.DEPTH_INFINITE);
-					String pageContents = getDocument().get();
-					Reader reader = new StringReader(pageContents);
+					String documentContent = getDocument().get();
 					/*
-					 * dummy is here to be able to suppress the warning about the
-					 * unused new Template()
+					 * dummy is here to be able to suppress the warning about
+					 * the unused new Template()
 					 */
-					@SuppressWarnings("unused")
-					Template dummy = new Template(getFile().getName(), reader,
-							fmConfiguration);
-					reader.close();
-				}
-			} catch (ParseException e) {
-				if (e.getMessage() != null) {
-					String errorStr = e.getMessage();
-					int errorLine = e.getLineNumber();
-					if (errorLine == 0) {
-						// sometimes they forget to put it in
-						int index = e.getMessage().indexOf("line: "); //$NON-NLS-1$
-						if (index > 0) {
-							int index2 = e.getMessage().indexOf(
-									" ", index + 6); //$NON-NLS-1$
-							int index3 = e.getMessage().indexOf(
-									",", index + 6); //$NON-NLS-1$
-							if (index3 < index2 && index3 > 0)
-								index2 = index3;
-							String s = e.getMessage().substring(index + 6,
-									index2);
-							try {
-								errorLine = Integer.parseInt(s);
-							} catch (NumberFormatException e2) {
-								errorLine = 0;
-							}
-						}
+					try {
+						@SuppressWarnings("unused")
+						Template dummy = new Template(getFile().getName(), documentContent, fmConfiguration);
+					} catch (ParseException e) {
+						int startLine = e.getLineNumber();
+						int startColumn = e.getColumnNumber();
+						int firstCharOffset = calculateOffset(documentContent, 0, 1, 1, startLine, startColumn);
```

Comment:
I guess I just haven't found that methods pack then. Thanks for pointing this out, change committed.